File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_distance_us import DistanceUS

# Callback function for distance callback (parameter has unit mm)
def cb_distance(distance):
    print('Distance: ' + str(distance/10.0) + ' cm')

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    dist = DistanceUS(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Set Period for distance callback to 0.2s (200ms)
    # Note: The callback is only called every 200ms if the 
    #       distance has changed since the last call!
    dist.set_distance_callback_period(200)

    # Register distance callback to function cb_distance
    dist.register_callback(dist.CALLBACK_DISTANCE, cb_distance)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'jAw'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var dist = new Tinkerforge.BrickletDistanceUS(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set Period for distance callback to 0.2s (200ms)
        // Note: The callback is only called every 200ms if the
        // distance has changed since the last call!
        dist.setDistanceCallbackPeriod(200);
    }
);

// Register distance callback
dist.on(Tinkerforge.BrickletDistanceUS.CALLBACK_DISTANCE,
    // Callback function for distance value
    function(distance) {
        console.log('Distance: '+distance);
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
